utils/settings.py

```diff
@@ -1,170 +1,260 @@
 
-def check_toml(template_file, config_file) -> Tuple[bool, Dict]:
+    Args:
+        path: List with path
+        checks: Dict with all checks
+    """
     global config
-    config = None
-    try:
-        template = toml.load(template_file)
-    except Exception as error:
-        console.print(f"[red bold]Encountered error when trying to to load {template_file}: {error}")
-        return False
-    try:
-        config = toml.load(config_file)
-    except toml.TomlDecodeError:
-        console.print(
-            f"""[blue]Couldn't read {config_file}.
-Overwrite it?(y/n)"""
-        )
-        if not input().startswith("y"):
-            print("Unable to read config, and not allowed to overwrite it. Giving up.")
+    if checks is None:
+        checks = dict()
+
+    value = check(
+        nested_get(config, path),
+        checks,
+        name=path[-1],
+    )
+    nested_set(config, path, value)
+
+
+def check_config_wrapper(
+        func: function,
+) -> function:
+    """
+    Exception wrapper for check_config function
+    """
+
+    def wrapper(*args, **kwargs):
+        if args:
+            kwargs["name"] = args[0]
+            if args.__len__() > 1:
+                kwargs["template_name"] = args[-1]
+        if not kwargs or not all(arg is not None for arg in kwargs.values()):
+            kwargs["name"] = config_name
+            kwargs["template_name"] = config_template_name
+
+        try:
+            return func(*args, **kwargs)
+        except toml.TomlDecodeError:
+            if console.input(f"[blue]Couldn't read {kwargs['name']}.\nOverwrite it?(y/n)").startswith("y"):
+                try:
+                    with open(kwargs["name"], "w") as f:
+                        f.write("")
+                    return func(*args, **kwargs)
+                except Exception:  # noqa (Exception is fine, it's not too broad)
+                    console.print(
+                        f"[red bold]Failed to overwrite {kwargs['name']}. Giving up.\n"
+                        f"Suggestion: check {kwargs['name']} permissions for the user."
+                    )
+                    return False
+            console.print("Unable to read config, and not allowed to overwrite it. Giving up.")
             return False
-        else:
-            try:
-                with open(config_file, "w") as f:
-                    f.write("")
-            except:
-                console.print(
-                    f"[red bold]Failed to overwrite {config_file}. Giving up.\nSuggestion: check {config_file} permissions for the user."
-                )
-                return False
-    except FileNotFoundError:
-        console.print(
-            f"""[blue]Couldn't find {config_file}
-Creating it now."""
-        )
+        # except Exception as error:
+        #     console.print(f"[red bold]Encountered error when trying to to load {kwargs['template_name']}: {error}")
+        #     return False
```

Comment:
forgot to uncomment this)
```suggestion
        except Exception as error:
            console.print(f"[red bold]Encountered error when trying to to load {kwargs['template_name']}: {error}")
            return False
```